extensions/mongoProcessor/MongoQueueProcessor.js

```diff
@@ -269,30 +368,52 @@ class MongoQueueProcessor {
+                    method: 'MongoQueueProcessor._processObjectQueueEntry',
+                    entry: sourceEntry.getLogInfo(),
+                });
+                return done(err);
+            }
+            // identify duplicate entry if the object key w/ version id already
+            // exists in mongo and the current entry is not a master key
+            if (zenkoObjMd && !this._isMasterKey(sourceEntry)) {
+                this.logger.debug('skipping duplicate entry', {
+                    method: 'MongoQueueProcessor._processObjectQueueEntry',
+                    entry: sourceEntry.getLogInfo(),
+                });
+                return process.nextTick(done);
+            }
+            // TODO: for md-only updates, content will need to be checked
+            //   based off previous entries
+            const content = this._getDataContent(sourceEntry);
 
-        const objVal = sourceEntry.getValue();
-        // Always call putObject with version params undefined so
-        // that mongoClient will use putObjectNoVer which just puts
-        // the object without further manipulation/actions.
-        // S3 takes care of the versioning logic so consuming the queue
-        // is sufficient to replay the version logic in the consumer.
-        return this._mongoClient.putObject(bucket, key, objVal, undefined,
-            this.logger, err => {
-                if (err) {
-                    this.logger.error('error putting object metadata ' +
-                    'to mongo', { error: err });
-                    return done(err);
-                }
-                this.logger.info('object metadata put to mongo',
-                { key });
-                return done();
-            });
+            // update necessary metadata fields before saving to Zenko MongoDB
+            this._updateOwnerMD(sourceEntry, bucketInfo);
+            this._updateObjectDataStoreName(sourceEntry, location);
+            this._updateLocations(sourceEntry, location);
+            this._updateAcl(sourceEntry);
+            this._updateReplicationInfo(sourceEntry, bucketInfo, content);
+
+            const objVal = sourceEntry.getValue();
+            // Always call putObject with version params undefined so
+            // that mongoClient will use putObjectNoVer which just puts
+            // the object without further manipulation/actions.
+            // S3 takes care of the versioning logic so consuming the queue
+            // is sufficient to replay the version logic in the consumer.
+            return this._mongoClient.putObject(bucket, key, objVal, undefined,
+                this.logger, err => {
+                    if (err) {
+                        this.logger.error('error putting object metadata ' +
+                        'to mongo', { error: err });
+                        return done(err);
+                    }
+                    this.logger.info('object metadata put to mongo',
+                    { key });
```

Comment:
Maybe valuable also to log the bucket name because there could be multiple of the same key names in MD across buckets.